core/src/main/java/org/bitcoinj/core/Peer.java

```diff
@@ -1552,6 +1557,10 @@ private void addPingTimeData(long sample) {
         final VersionMessage ver = vPeerVersionMessage;
         if (!ver.isPingPongSupported())
             throw new ProtocolException("Peer version is too low for measurable pings: " + ver);
+        if (pendingPings.size() > 50) {
```

Comment:
You're fiixing two different cases. I'd prefer to have them in two commits, one for each.